Index: xplayer-3.4.2/data/Makefile.am
===================================================================
--- xplayer-3.4.2.orig/data/Makefile.am	2012-05-15 19:33:09.509491282 -0400
+++ xplayer-3.4.2/data/Makefile.am	2012-05-15 19:33:11.465500977 -0400
@@ -58,7 +58,7 @@
 
 xplayer.desktop.in: xplayer.desktop.in.in mime-type-list.txt uri-schemes-list.txt desktop.sh
 	$(AM_V_GEN) cat xplayer.desktop.in.in | sed 's,@FULL_LIBEXECDIR@,$(FULL_LIBEXECDIR),' > $@ &&\
-	$(SHELL) $(srcdir)/desktop.sh $(srcdir)/mime-type-list.txt $(srcdir)/uri-schemes-list.txt >> $@
+	$(AM_V_GEN) cat xplayer.desktop.in.in | sed 's,@MIMETYPES@,$(shell $(srcdir)/desktop.sh $(srcdir)/mime-type-list.txt $(srcdir)/uri-schemes-list.txt),' > $@
 
 EXTRA_DIST += desktop.sh
 CLEANFILES += \
Index: xplayer-3.4.2/data/xplayer.desktop.in.in.in
===================================================================
--- xplayer-3.4.2.orig/data/xplayer.desktop.in.in.in	2012-05-15 19:33:09.509491282 -0400
+++ xplayer-3.4.2/data/xplayer.desktop.in.in.in	2012-05-15 19:33:11.465500977 -0400
@@ -6,6 +6,7 @@
 Terminal=false
 Type=Application
 Categories=GTK;GNOME;AudioVideo;Player;Video;
+@MIMETYPES@
 X-GNOME-DocPath=xplayer/xplayer.xml
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=xplayer
